packages/shorebird_cli/lib/src/commands/patch/patcher.dart

```diff
@@ -115,4 +115,42 @@ https://docs.shorebird.dev/status#link-percentage-ios
   File get debugInfoFile {
     return File(p.join(buildDirectory.path, 'patch-debug.zip'));
   }
+
+  /// Extracts the --build-name and --build-number from the --release-version
+  /// argument if it's provided. Given `--release-version=1.2.3+4`, this will
+  /// return `['--build-name=1.2.3', '--build-number=4']`, with the intent that
+  /// these values will be forwarded to the `flutter build` command.
+  ///
+  /// Because not all platform types support both --build-name and
+  /// --build-number, this needs to be handled in the platform-specific
+  /// patchers instead of at the patch command level.
+  ///
+  /// We do this because some platforms encode the build version in their
+  /// binaries (Android does this with .dex files). If a release and a patch
+  /// have different version numbers, our [PatchDiffChecker] to warn the user of
+  /// native changes, even though the user may not have actually changed any
+  /// code or dependencies.
+  ///
+  /// Context: https://github.com/shorebirdtech/shorebird/issues/2270
+  List<String> buildNameAndNumberArgsFromReleaseVersion(
+    String? releaseVersion,
+  ) {
+    if (releaseVersion == null || !releaseVersion.contains('+')) {
```

Comment:
Is a release version guaranteed to have a build number?

Reply:
I think so, but was being defensive here.

Reply:
> what happens if a user includes version: 1.0.0 in their pubspec.yaml? Might be another source of confusion

I believe both Android and iOS will use the version as the build number (so `1.0.0` becomes `1.0.0+1.0.0`). I think this is sort of tangential to the issue here—this PR doesn't change how the release version is parsed, it just tries to build the patch with the same version as the release.

Reply:
Correction: this is the behavior on iOS, but Android uses `1` for the build number if none is provided in the pubspec